### PR TITLE
Allow DLLs

### PR DESCRIPTION
**Pull request by jwaldrop107**
**Is this pull request associated with an issue(s)?**
None

**Description**
Lifts the restriction on libraries having the suffix `dll`.

### CODE REVIEW

File: cmake/cmaize/cmaize.cmake [hunk 1]
```diff
@@ -38,3 +38,11 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
...
 
 list(APPEND CMAKE_INSTALL_RPATH "$ORIGIN")
+
+# possibly a hack, but allows the test exes to find the dlls
+# if (WIN32)
+#     set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+#     set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+#     set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+#     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+# endif (WIN32)
```
Comment (ryanmrichard) [inline cmake/cmaize/cmaize.cmake:48]: Is this still needed or should it be deleted
```suggestion
```

Comment (jwaldrop107) [reply cmake/cmaize/cmaize.cmake:48]: I left it mostly as a note. I can remove them, but I figured leave them as reference for later attempts at a more complete windows solution. @zachcran and I talked about it and didn't want to add these global settings for now. But someone else could use these options as a holdover for now if they so choose.

**Comment by ryanmrichard** [comment]
Once @jwaldrop107 responds to my review I'm willing to force merge this. I opened #161 to track the CI problem.

File: cmake/cmaize/cmaize.cmake (hunk 1 above)
Comment (ryanmrichard) [reply cmake/cmaize/cmaize.cmake:48]: Gotcha. Leave them in then.

**Pull request merged**